src/main/app/controller/courts/CourtTypesController.ts

```diff
@@ -24,64 +30,79 @@ export class CourtTypesController {
     res: Response,
     updated = false,
     error = '',
-    courtCourtTypes: CourtType[] = null): Promise<void> {
-    if (!courtCourtTypes) {
-
-      const slug: string = req.params.slug as string;
-      await req.scope.cradle.api.getCourtCourtTypes(slug)
-        .then((value: CourtType[]) => courtCourtTypes = value)
-        .catch(() => error += this.getCourtTypesErrorMsg);
-
+    courtTypesAndCodes: CourtTypesAndCodes = null ): Promise<void> {
+    const slug: string = req.params.slug as string;
+    if (courtTypesAndCodes == null) {
```

Comment:
Would it be better to declare it as 

courtTypesAndCodes: CourtTypesAndCodes = []

and then do 

if (!courtTypesAndCodes.length) 

As that then means we are not passing nulls around, and instead are working with lists 

Thoughts ? :)